chore: improve changelog generation (#204)
biomejs/biome: CONTRIBUTING.md

### Creating pull requests

When creating a new pull request, it's preferable to use a conventional commit-formatted title, as this title will be used as the default commit message on the squashed commit after merging.
See the [dedicated section](#Commit-messages) about conventional commit format.

Please use the template provided.

#### Changelog

> If the PR you're about to open is a bugfix/feature that is visible to Biome users,
> you CAN add a new bullet point to [CHANGELOG.md](./CHANGELOG.md).
> Although **not required**, we appreciate the effort.

Reviewer: ```suggestion
If the PR you're about to open is a bugfix/feature visible to Biome users, you CAN add a new bullet point to [CHANGELOG.md](./CHANGELOG.md). Although **not required**, we appreciate the effort.
```

At the top of the file you will see a `Unreleased` section.

> The headings divide the sections by "scope".
> Make sure to edit the right subsection.

Reviewer: ```suggestion
The headings divide the sections by "scope"; you should be able to identify the scope that belongs to your change. If the change belongs to multiple scopes, you can copy the same sentence under those scopes.
```

- If your PR adds a **breaking change**, create a new heading called `#### BREAKING CHANGES` and add
  bullet point that explains the breaking changes; provide a migration path if possible.
  Read [how we version Biome](https://biomejs.dev/internals/versioning/) to determine if your change is a breaking change.
  A breaking change result in a major release.
- If your PR adds a new feature, enhance an existing feature, or fix a bug, create a new heading called `#### New features`, `#### Enhancements`, or `#### Bug fixes`.
  Then, add a bullet point that explains the change.

Make sure that the created subsections are ordered in the following order:

```md
#### BREAKING CHANGES

#### New features

> #### Enhancements

Reviewer: What's the difference between enhancements and features?

Author: `enhancements` improve an existing feature (for example extending cases covered by a rule). `New features` includes new added rules, new config options, ...

- If your PR adds a **breaking change**, create a new heading called `#### BREAKING CHANGES` and add
  bullet point that explains the breaking changes; provide a migration path if possible.

>   Read [how we version Biome](https://biomejs.dev/internals/versioning/) to determine if your change is a breaking change.
>   A breaking change result in a major release.
> - If your PR adds a new feature, enhance an existing feature, or fix a bug, create a new heading called `#### New features`, `#### Enhancements`, or `#### Bug fixes`.
>   Then, add a bullet point that explains the change.
> 
> Make sure that the created subsections are ordered in the following order:

Reviewer: ```suggestion
  Read [how we version Biome](https://biomejs.dev/internals/versioning/) to determine if your change is breaking. A breaking change results in a major release.
- If your PR adds a new feature, enhances an existing feature, or fixes a bug, create a new heading called `#### New features`, `#### Enhancements`, or `#### Bug fixes`. Ultimately, add a bullet point that explains the change.

Make sure that the created subsections are ordered in the following order:
```

```md
#### BREAKING CHANGES

#### New features

#### Enhancements

#### Bug fixes
```

> Bullet points should be sorted in their importance order: important changes come first.

Reviewer: Who decides the importance of a fix? How is it decided? "Why is my fix less important than yours"? I don't think we should add this.

> Because the website displays the changelog,
> you should update the website using the following command:

Reviewer: ```suggestion
Because the website displays the changelog, you should update the website using the following command:
```

```sh

> cargo codegen-website

Reviewer: ```suggestion
just codegen-website
```

Let's use `just`; that's the tool we're telling contributors to use. If we don't have a script for that, let's create one

### Playground

- [run the playground locally](/website/playground/README.md)

## Releasing

When releasing a new version of a Biome, follow these steps:

1. [ ] Add a [changelog](./CHANGELOG.md) entry for every Pull Request that lacks one.
   You can filter [merged PRs that don't update the changelog](https://github.com/biomejs/biome/pulls?q=is%3Apr+is%3Amerged+-label%3AA-Changelog).
   Read our [guidelines for editing the changelog](#changelog).

1. [ ] Based on the [changelog](./CHANGELOG.md), determine which version number to use.
   See our [versioning guide](https://biomejs.dev/internals/versioning/) for more details.

> 1. [ ] Rename `Unreleased` to `<version> (iso-date)` in the [changelog](./CHANGELOG.md).
>    Then update the website using `cargo codegen-website`.

Reviewer: ```suggestion
1. [ ] Rename `Unreleased` to `<version> (iso-date)` in the [changelog](./CHANGELOG.md).
   Then update the website using `BIOME_VERSION=<version> cargo codegen-website`.
```

This is important, with `BIOME_VERSION`, the task will generate the new JSON schema

> 1. [ ] Update `version` in [Biome's `package.json`](./packages/@biomejs/biome/package.json).

Reviewer: ```suggestion
1. [ ] Update `version` in [Biome's `package.json`](./packages/@biomejs/biome/package.json) if applicable.
```

> 1. [ ] Update `version` in [Biome's LSP package.json](./editors/vscode/package.json).

Reviewer: ```suggestion
1. [ ] Update `version` in [Biome's LSP package.json](./editors/vscode/package.json) if applicable.
```

   Note that the LSP follows a [distinct versioning scheme](https://biomejs.dev/internals/versioning/#visual-studio-code-extension).

> 1. [ ] Linter rules has a `version` metadata directly defined in their implementation.

Reviewer: ```suggestion
1. [ ] Linter rules have a `version` metadata directly defined in their implementation.
```

>    This field is set to `next` for newly added rules.

Reviewer: ```suggestion
   This field is set to `next` for newly created rules.
```

   This field must be updated to the new version.
   Then execute `just codegen-linter`.

1. [ ] Biome's website exposes the JSON schema for `biome.json`.
   Copy the [up-to-date JSON schema](./packages/@biomejs/biome/configuration_schema.json) to `website/src/pages/schemas/<version>/schema.json.js`.

>    Don't forget to replace `<version>` in the path with the chosen version.

Reviewer: We can remove this step, it should be covered when we run `cargo codegen-website`
